stdlib/string.ml

```diff
@@ -231,3 +231,22 @@ let to_seq s = bos s |> B.to_seq
 let to_seqi s = bos s |> B.to_seqi
 
 let of_seq g = B.of_seq g |> bts
+
+let starts_with ~prefix s =
+  let len_s = length s
+  and len_pre = length prefix in
+  let rec aux i =
```

Comment:
I believe this will allocate a closure
extlib uses the following implementation : https://github.com/ygrek/ocaml-extlib/blob/master/src/extString.ml#L43